Geometry/HGCalGeometry/src/FastTimeGeometry.cc

```diff
@@ -125,7 +125,7 @@ DetId FastTimeGeometry::getClosestCell(const GlobalPoint& r) const {
     double zz = (zside > 0) ? r.z() : -r.z();
     etaZPhi = topology().dddConstants().getZPhi(zz,r.phi());
   } else {
-    double phi = (zside > 0) ? r.phi() : atan2(r.y(),-r.x());
+    double phi = (zside > 0) ? static_cast<double>(r.phi()) : atan2(r.y(),-r.x());
```

Comment:
What is the purpose of explicitly adding this ```static_cast``` (and only here, not in L126, e.g.)?

Reply:
The conditional operator `?` is resolved at compile time. It requires its two results to be the same type. Because the `Phi` class has various conversions and `atan2` is a template function, there is a compile-time ambiguity in the desired type of the results, though it is obvious to the reader. The static cast eliminates the ambiguity for the compiler.

This problem only occurs with the conditional operator, and probably only when the two results of the operator are classes or templates that can take on various types. If this statement were written as an if-then statement, there would be no problem. I checked out all the dependencies that use the `Phi` class and compiled all that code, and this line of code was the only one where there was any type ambiguity.

Everywhere else in the code, variables of the `Phi` class are automatically converted into the correct type at execute time, as far as I can tell.

I will add a comment here to explain the cast.